services/youtube/youtube.service.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService, NotFound } = require('..')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schema = Joi.object({
+  items: Joi.array()
+    .items(
+      Joi.object({
+        statistics: Joi.object({
+          viewCount: nonNegativeInteger,
+          likeCount: nonNegativeInteger,
+          dislikeCount: nonNegativeInteger,
+          commentCount: nonNegativeInteger,
+        }).required(),
+      })
+    )
+    .required(),
+}).required()
+
+module.exports = class YouTube extends BaseJsonService {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'youtube',
+      pattern: ':statistic(view|like|dislike|comment)/:videoId',
```

Comment:
Could I suggest instead of a dislikes badge, we do a query param variant on the likes badge, that shows both likes and dislikes?

**votes | 2.5k up, 100 down**

It's a little like the way we do failing tests.

Reply:
Good idea, will implement sometime this week!